Dmitriy_Vysotskyy2.0/PageObjects/HomePage.cs

```diff
@@ -0,0 +1,88 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.PageObjects;
+using SeleniumExtras.WaitHelpers;
+
+namespace Dmitriy_Vysotskyy2._0.PageObjects;
+
+public class HomePage : BasePage
+{
+    private string _urlLink = "https://www.demoblaze.com";
+
+    public HomePage(IWebDriver driver) : base(driver)
+    {
+       
+    }
+    
+    [FindsBy(How = How.Id, Using = "signin2")]
+    [CacheLookup]
+    private IWebElement _btnOpenSignUp;
+    
+    [FindsBy(How = How.Id, Using = "login2")]
+    [CacheLookup]
+    private IWebElement _btnOpenLogIn;
+    
+    [FindsBy(How = How.Id, Using = "sign-username")]
+    [CacheLookup]
+    private IWebElement _txtUsernameSignUp;
+
+    [FindsBy(How = How.Id, Using = "sign-password")]
+    [CacheLookup]
+    private IWebElement _txtPasswordSignUp;
+    
+    [FindsBy(How = How.Id, Using = "loginusername")]
+    [CacheLookup]
+    private IWebElement _txtUsernameLogIn;
+
+    [FindsBy(How = How.Id, Using = "loginpassword")]
+    [CacheLookup]
+    private IWebElement _txtPasswordLogIn;
+    
+    [FindsBy(How = How.XPath, Using = "//button[contains(@class, 'btn btn-primary') and text()='Log in']")]
+    [CacheLookup]
+    private IWebElement _btnLogIn;
+    
+    [FindsBy(How = How.XPath, Using = "//button[contains(@class, 'btn btn-primary') and text()='Sign up']")]
+    [CacheLookup]
+    private IWebElement _btnSignUp;
+    
+    public void Navigate()
+    {
+        _driver.Navigate().GoToUrl(_urlLink);
+        _driver.Manage().Window.Maximize();
+    }
+    
+    public string SignUp(string username, string password)
+    {
+        _wait.Until(ExpectedConditions.ElementExists(By.Id("signin2"))); //DRY is out of party
+        _btnOpenSignUp.Click();
+        
+       // _wait.Until(ExpectedConditions.ElementExists(By.XPath("/html/body/div[2]/div/div/div[3]/button[1]"))); //DRY is out of party
```

Comment:
To avoid locators duplocation you can put its value into some variable on the class level and use both in WebElement and in waiter OR create your custom WebElement class and store it inside